src/main/java/com/openlattice/shuttle/dates/JavaDateTimeHelper.java

```diff
@@ -1,88 +1,94 @@
 package com.openlattice.shuttle.dates;
 
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.openlattice.shuttle.util.Cached;
 import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.text.SimpleDateFormat;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
 import java.time.format.DateTimeFormatter;
-import java.util.Arrays;
-import java.util.List;
+import java.time.format.DateTimeParseException;
 import java.util.TimeZone;
-import java.util.stream.Collectors;
+import java.util.concurrent.ExecutionException;
+import java.util.function.BiFunction;
 
 public class JavaDateTimeHelper {
     private static final Logger logger = LoggerFactory.getLogger(JavaDateTimeHelper.class);
 
-    private final TimeZone tz;
-    private final String[] datePatterns;
-    private final List<DateTimeFormatter> formatters;
+    private final TimeZone      tz;
+    private final String[]      datePatterns;
 
     public JavaDateTimeHelper(TimeZone tz, String... datePatterns) {
         this.tz = tz;
         this.datePatterns = datePatterns;
-        this.formatters = Arrays.stream(datePatterns).map(pattern -> DateTimeFormatter.ofPattern(pattern))
-                .collect(Collectors.toList());
+    }
+
+    private static LoadingCache<String, DateTimeFormatter> buildCache() {
```

Comment:
is this ever called?

Reply:
Derp, leftovers from previous iteration, will remove